### PR TITLE
[PLATFORM-1307] Publish dialog refactor

### PR DESCRIPTION
**Pull request by juhah**
Publish dialog refactor, using a progress bar to indicate progress.

### CODE REVIEW

**Comment by juhah** [comment]
@mattatgit PR url: http://streamr-marketplace-pr-d33f816f9e45de5f2022d0a8dd014039de04ebff.s3-website-eu-west-1.amazonaws.com

**Comment by mattatgit** [comment]
@juhah 

Here are my notes from testing it out a few times

* still shows old spinner for connecting prior to Publish dialogue being shown
* spacing of Cancel beside button seems too wide (right edge of "Cancel" text box should be 24px from left edge of main action button)
* padding of "Copy" text inside Completed dialogue field looks a bit off
* if it's going to show each time, perhaps instead of "Requires wallet confirmation" in orange, we just keep it in grey (as not really an error) and say "Please confirm transaction". Seems a bit wrong otherwise.
* unpublishing gave an error, didn't work
* Publish & unpublish progress bars did not show at all for the tests i published. Even if there is very little to do, the bar should still animate across, or it seems broken.

Here's a link to a screencapture video of one publish

https://www.dropbox.com/s/o4w7b9c67aywtne/Publish%20flow.mp4?dl=0

**Comment by mattatgit** [comment]
re the old spinner, is this something that only shows on slow connections, like mine thru the VPN ?
Or does everyone see that while it connects to user's wallet ? I think we'll want to get rid of / update that, can be a separate issue, but doesn't look to good so would like to eliminate or update it.

**Comment by mattatgit** [comment]
Screenshot 2020-03-31 11 37 30

If we update the Terms section to the above, and make it a required section, we can just drop the entire first slide of that process & go straight to the publish flow.

@juhah I've already added this the Core Abstract, so can make a ticket for it if you want.

**Comment by juhah** [comment]
@mattatgit 
> re the old spinner, is this something that only shows on slow connections, like mine thru the VPN ?
Or does everyone see that while it connects to user's wallet ? I think we'll want to get rid of / update that, can be a separate issue, but doesn't look or feel too good so would like to eliminate or update it — It really makes the process feel slow ie having to wait just to see the actual publish dialogue.

This spinner is in all dialogs, it's shown while loading the product info (eg. here and the purchase dialog). I'm not sure which is the new spinner design but you could make a ticket about it and we can handle it separately for all dialogs.

> If we update the Terms section to the above, and make it a required section, we can just drop the entire first slide of that process & go straight to the publish flow, making it much more logical & shorter.

We'll need to update this to the backend so that will take time to implement (adding the terms & contact details is also still open). I would suggest either:

- continuing with the current design, or
- abandon this and implement the new design when backend is ready

**Comment by mattatgit** [comment]
@juhah Ok, we should continue with this design for now to avoid holdup, but we should circle back and improve it after this pass is done.

Re spinner, there isn't a global one yet, this publish flow update is aimed partly at removing it. But let's address that in a separate ticket

**Comment by juhah** [comment]
@mattatgit ok, i wil finish this and let's improve on the publish confirmation later.

> * Publish & unpublish progress bars did not show at all for the tests i published. Even if there is very little to do, the bar should still animate across, or it seems broken.
> * padding of "Copy" text inside Completed dialogue field looks a bit off
> * if it's going to show each time, perhaps instead of "Requires wallet confirmation" in orange, we just keep it in grey (as not really an error) and say "Please confirm transaction". Seems a bit wrong otherwise.

I've address these, the new PR url is:  http://streamr-marketplace-pr-933442578c200b8d9816c88b843790852226b08e.s3-website-eu-west-1.amazonaws.com
4009

These issues affect all modals and should be fixed at once:

> * still shows old spinner for connecting prior to Publish dialogue being shown
> * the final slide in the flow is missing the grey horizontal divider above the main action button
> * spacing of Cancel beside button seems too wide (right edge of "Cancel" text box should be 24px from left edge of main action button)

Cancel buttons are shown as buttons that are same widths as the others buttons across all dialogs, hence the spacing issue.

I added issue https://streamr.atlassian.net/browse/PLATFORM-1315 for this, feel free to update it with more things.

**Comment by juhah** [comment]
Inviting some reviews from @tumppi @mondoreale, please have a look. Tests are failing in staging but locally they are working fine 🤔 If you run the tests locally and figure out why, please let me know.

**Comment by mondoreale** [comment]
@juhah, found a solution to the test issue, see https://github.com/streamr-dev/streamr-platform/pull/947/commits/8e077acf2604013926f14ee295ead7f71f9d1b9e.

I took a fairly weird path to get there:
https://github.com/streamr-dev/streamr-platform/compare/99ad0a6...8e077ac

💃

**Comment by mondoreale** [comment]
Already pushed in here, 21a6c8b. Hope you don't mind me interfering with your work. ;)

**Comment by mondoreale** [comment]
@juhah can you go through the chromatic build and accept/reject the changes too?

**Comment by juhah** [comment]
@mondoreale 
> Already pushed in here, 21a6c8b. Hope you don't mind me interfering with your work. ;)

Amazing, thanks for figuring it out! I definitely owe you a 🍺 

> can you go through the chromatic build and accept/reject the changes too?

Yep, should be ok. The publish dialog looks off for some reason in the pictures but storybook looks ok locally.

**Comment by juhah** [comment]
PR url: http://streamr-marketplace-pr-21a6c8bac32e418c9bb30e6970d65b6f847bc977.s3-website-eu-west-1.amazonaws.com

File: app/src/editor/shared/tests/ModuleSubscription.test.jsx [hunk 1]
```diff
@@ -91,7 +91,7 @@ describe('Canvas Subscriptions', () => {
             done()
         }, 15000)
 
-        it('should get canvas module subscription messages in restarted canvas', async (done) => {
+        xit('should get canvas module subscription messages in restarted canvas', async (done) => {
```
Comment (timoxley) [inline app/src/editor/shared/tests/ModuleSubscription.test.jsx:94]: perhaps no longer needed with https://github.com/streamr-dev/streamr-platform/pull/952

**Comment by mattatgit** [comment]
Whenever i try to create a product, i just get this error view @juhah 
Screenshot 2020-04-08 15 54 19

**Comment by juhah** [comment]
@mattatgit seems to be an issue with PR build, I'll merge this and address the feedback in https://streamr.atlassian.net/browse/PLATFORM-1329

**Pull request merged**